src/reanimated2/hook/utils.ts

```diff
@@ -201,12 +201,12 @@ export function isAnimated(prop: NestedObjectValues<AnimationObject>): boolean {
   return false;
 }
 
-export function styleDiff(
+export function styleDiff<T extends AnimatedStyle>(
   oldStyle: AnimatedStyle,
   newStyle: AnimatedStyle
-): AnimatedStyle {
+): T {
   'worklet';
-  const diff: AnimatedStyle = {};
+  const diff: any = {};
```

Comment:
Could be something like:
```suggestion
  const diff: Partial<T> = {};
```

Reply:
Fixed with [8da3ad0](https://github.com/software-mansion/react-native-reanimated/pull/2776/commits/8da3ad0094bf0c86ce697f92993462278a90c163)